src/tuwien/auto/calimero/knxnetip/Discoverer.java

```diff
@@ -644,8 +644,9 @@ public final void clearSearchResponses()
 					: nat ? new InetSocketAddress(0) : new InetSocketAddress(localAddr, s.getLocalPort());
 
 			// send search request with additional DIBs
-			final byte[] extraDibs = PacketHelper.toPacket(new SearchRequest(res, DIB.DEVICE_INFO, DIB.SUPP_SVC_FAMILIES,
-					DIB.AdditionalDeviceInfo, DIB.SecureServiceFamilies, DIB.Tunneling));
+			final byte[] extraDibs = PacketHelper.toPacket(new SearchRequest(res,
+				new SRP(SRP.REQUEST_DIBS, false, (byte) DIB.DEVICE_INFO, (byte) DIB.SUPP_SVC_FAMILIES,
+					(byte) DIB.AdditionalDeviceInfo, (byte) DIB.SecureServiceFamilies, (byte) DIB.Tunneling)));
```

Comment:
I find it rather odd not being able to set the SRPs from the API. I will propose a setter or a ctor if you think that makes sense.

Reply:
That the new search request with the request DIBs block is there "as-is" is on purpose, because I (currently) think that it provides a sensible default for finding nodes for a user, without having to know too much about the recent changes. You are right, that the search parameters should be configurable in general.

Reply:
My thoughts on this: when I added the search.req ctor with requestDibs param, I played around with some "nicely named" static methods to create the search parameters. Since there is now the SRP class, this would be nice to do in that class. So to have a configuration à la 
`Discoverer::select(inProgrammingMode(), withMacAddress(...))`

where the configuration could go in the ctor or in some metod like `select/request/whatever`. The static factory method names should be intuitive for a user, assuming a static import from SRP. 
What do you think?

Reply:
This sounds very reasonable. I will try to come up with something in the next patch-set.

Reply:
I did an initial implementation in a second commit. Hope this gets in the right direction.